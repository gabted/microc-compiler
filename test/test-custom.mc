char s[] = "hello world!";
double a[][10];
double (*p)[][10] = &a;
bool *q = NULL;

void print_matrix(int n){
    for(int i = 0; i<n; i++)
        for(int j=0; j<10; j++)
            print_double(a[i][j]);
}

void print_string(){
    for(int i = 0; s[i] != '\0'; i++)
        print_char(s[i]);
    print_char('\n');//prints two newlines
}

void main(){
    print_string();
    for(int i = 0; i<5; i++)
        for(int j=0; j<10; j++)
            a[i][j] = (i+1)*(j+1)/2.;
    print_matrix(5);
}