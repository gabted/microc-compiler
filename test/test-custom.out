h
e
l
l
o
 
w
o
r
l
d
!


0.500000
1.000000
1.500000
2.000000
2.500000
3.000000
3.500000
4.000000
4.500000
5.000000
1.000000
2.000000
3.000000
4.000000
5.000000
6.000000
7.000000
8.000000
9.000000
10.000000
1.500000
3.000000
4.500000
6.000000
7.500000
9.000000
10.500000
12.000000
13.500000
15.000000
2.000000
4.000000
6.000000
8.000000
10.000000
12.000000
14.000000
16.000000
18.000000
20.000000
2.500000
5.000000
7.500000
10.000000
12.500000
15.000000
17.500000
20.000000
22.500000
25.000000
